src/NovelRT/Ecs/SparseSetMemoryContainer.cpp

```diff
@@ -82,24 +82,29 @@ namespace NovelRT::Ecs
 
         _sparse[key] = 0;
 
-        bool canResize = true;
         if (!_dense.empty())
         {
-            for (auto it = _sparse.begin() + key; it != _sparse.end(); it++)
+            std::optional<std::vector<size_t>::iterator> targetStart{};
+            size_t currentValue = 0;
+            size_t finalDenseSize = _dense.size();
+            for (auto it = _sparse.rbegin() + key; it != _sparse.rend(); it++)
```

Comment:
This is going to do `key` elements from the *end*, rather than `key` elements from the *beginning*.